From: Robin Snyders <robin@snyders.xyz>
Date: Wed, 12 Aug 2026 00:00:00 +0200
Subject: [PATCH] arm64: dts: qcom: hotdog: feed the NFC reference clock

The controller has no crystal of its own and asks the platform for its
reference on the clock request line. Point it at LN_BB_CLK3, which is the
19.2 MHz buffer this board leaves unclaimed.

Signed-off-by: Robin Snyders <robin@snyders.xyz>
---
diff --git a/arch/arm64/boot/dts/qcom/sm8150-oneplus-hotdog.dts b/arch/arm64/boot/dts/qcom/sm8150-oneplus-hotdog.dts
index 5039a42a9..9379eb304 100644
--- a/arch/arm64/boot/dts/qcom/sm8150-oneplus-hotdog.dts
+++ b/arch/arm64/boot/dts/qcom/sm8150-oneplus-hotdog.dts
@@ -10,6 +10,7 @@
 /dts-v1/;
 
 #include <dt-bindings/regulator/qcom,rpmh-regulator.h>
+#include <dt-bindings/clock/qcom,rpmh.h>
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/input/input.h>
 #include <dt-bindings/media/video-interfaces.h>
@@ -553,6 +554,7 @@ nfc@28 {
 
 		enable-gpios = <&tlmm 41 GPIO_ACTIVE_HIGH>;
 		firmware-gpios = <&tlmm 48 GPIO_ACTIVE_HIGH>;
+		clocks = <&rpmhcc RPMH_LN_BB_CLK3>;
 
 		pinctrl-names = "default";
 		pinctrl-0 = <&nfc_default>;
